scene/2d/tile_map.cpp

```diff
@@ -1163,8 +1205,7 @@ void TileMap::_physics_create_quadrant(TileMapQuadrant *p_quadrant) {
 	// Create the body and set its parameters.
 	for (int layer = 0; layer < tile_set->get_physics_layers_count(); layer++) {
 		RID body = PhysicsServer2D::get_singleton()->body_create();
-		PhysicsServer2D::get_singleton()->body_set_mode(body, PhysicsServer2D::BODY_MODE_STATIC);
-
+		PhysicsServer2D::get_singleton()->body_set_mode(body, PhysicsServer2D::BODY_MODE_KINEMATIC);
```

Comment:
Maybe the body type could be Kinematic only when sync to physics is enabled, otherwise by default changing the transform for the tilemap will collide with everything on the path.